src/groupeddataframe/grouping.jl

```diff
@@ -63,6 +69,12 @@ See the following for additional split-apply-combine operations:
 * `map` : apply a function to each group of a `GroupedDataFrame` (without combining)
 * `combine` : combine a `GroupedDataFrame`, optionally applying a function to each group
 
+`GroupedDataFrame` also supports the dictionary interface, where its keys are
+`NamedTuple`s containing the values of the grouped columns. When indexing using
+a `NamedTuple` key the fields may be in any order. You may also index using a
```

Comment:
do you find "any order" a valuable feature here. I usually tend to think that we should require a strict order (but opinions are mixed here 😄).

Reply:
I'm not sure if its a super valuable feature, but it was easy enough to code and I don't think creates too much overhead.

Reply:
We are strict not because of overhead but safety (i.e. if one passes column names in a wrong order then we are not sure if one would want to get a result or rather an error).

Reply:
I agree, better be strict. If we find out it's really annoying in practice we can be more tolerant in the future, but let's keep things simple and strict for now.